mods/modern-notepad-to-classic-editor.wh.cpp

```diff
@@ -0,0 +1,429 @@
+    std::wstring a = cur, b = classicPath;
+    for (auto &c : a) c = towlower(c);
+    for (auto &c : b) c = towlower(c);
+    if (a == b) {
+      InterlockedExchange(&g_handoffFinished, 1);
+      return 0;
+    }
+  }
+
+  // Elevate with Ctrl+Shift if enabled and not elevated
+  if (g_ctrlShiftElevate && IsCtrlShiftPressed() && !IsProcessElevated()) {
+    SHELLEXECUTEINFOW sei = { sizeof(sei) };
+    sei.lpVerb = L"runas";
+    sei.lpFile = classicPath.c_str();
+    std::wstring argTail = BuildArgsOnly();
+    sei.lpParameters = argTail.c_str();
+    sei.nShow = SW_SHOWNORMAL;
+
+    if (ShellExecuteExW(&sei)) {
+      if (g_waitForClassic && sei.hProcess) {
+        WaitForSingleObject(sei.hProcess, INFINITE);
+        DWORD exitCode = 0;
+        GetExitCodeProcess(sei.hProcess, &exitCode);
+        CloseHandle(sei.hProcess);
+        InterlockedExchange(&g_handoffFinished, 1);
+        ExitProcess(exitCode);
+      } else {
+        if (sei.hProcess) CloseHandle(sei.hProcess);
+        InterlockedExchange(&g_handoffFinished, 1);
+        ExitProcess(0);
+      }
+    }
+    // If elevation failed, fall through to non-elevated launch
+  }
+
+  // Non-elevated launch: build full command line with classic exe + original args
+  std::wstring newCmd = L"\"" + classicPath + L"\"";
+  {
+    std::wstring argTail = BuildArgsOnly();
+    if (!argTail.empty()) {
+      newCmd += L" ";
+      newCmd += argTail;
+    }
+  }
+
+  std::unique_ptr<wchar_t[]> buf(new wchar_t[newCmd.size() + 1]);
+  wcscpy_s(buf.get(), newCmd.size() + 1, newCmd.c_str());
+
+  STARTUPINFOW si = { 0 }; si.cb = sizeof(si);
+  PROCESS_INFORMATION pi = { 0 };
+
+  if (!CreateProcessW(NULL, buf.get(), NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi)) {
+    InterlockedExchange(&g_handoffFinished, 1);
+    ExitProcess(0);
+  }
+
+  if (g_waitForClassic) {
+    WaitForSingleObject(pi.hProcess, INFINITE);
+    DWORD exitCode = 0;
+    GetExitCodeProcess(pi.hProcess, &exitCode);
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+    InterlockedExchange(&g_handoffFinished, 1);
+    ExitProcess(exitCode);
+  } else {
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+    InterlockedExchange(&g_handoffFinished, 1);
+    ExitProcess(0);
+  }
+
+  return 0;
+}
+
+// ---------------------------------------------------------------------
+// Windhawk entry points
+// ---------------------------------------------------------------------
+BOOL Wh_ModInit() {
+  // Load settings
+  {
+    std::wstring pathSetting = Wh_GetStringSetting(L"classicPath");
+    if (pathSetting.empty()) pathSetting = L"%SystemRoot%\\System32\\notepad.exe";
+    g_classicPathSetting = pathSetting;
+
+    g_ctrlShiftElevate = Wh_GetIntSetting(L"ctrlShiftElevate", 1) != 0;
+    g_waitForClassic   = Wh_GetIntSetting(L"waitForClassic", 1) != 0;
+  }
+
+  // Only act inside the WindowsApps Notepad process
+  std::wstring cur = GetCurrentExePath();
+  if (!IsModernWindowsAppsNotepadPath(cur)) {
```

Comment:
Isn't this check redundant since the mod only targets the modern Notepad already?

Reply:
Yes, at some point I had just `notepad.exe` because I didn't want to hardcode the program files path and didn't realize environment variables work there.